PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiverAsyncClientCheck.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.NotNull;
+
+
+/**
+ * This class extends the LocalInspectionTool to check for the use of ServiceBusReceiverAsyncClient
+ * in the code and suggests using ServiceBusProcessorClient instead.
+ */
+public class ServiceBusReceiverAsyncClientCheck extends LocalInspectionTool {
+
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+
+        return new JavaElementVisitor() {
```

Comment:
AFAIK, we don't need to use anonymous classes. you can use a lambda.

Reply:
hmm I'm not sure if I can replace JavaElementVisitor with a lambda expression? I can look into this for future PRs

Reply:
Using a named class. Also makes testing easier.  https://gist.github.com/conniey/5e188d78102a789ef3901cb50e014039

Reply:
@Njeriiii ^^ This is a good coding practice I suggest you look into incorporating this in your PRs.